From 6e39f79a760f66bb5fded59f505b18c877cc3200 Mon Sep 17 00:00:00 2001
From: Devin Moore <devinmoore@google.com>
Date: Fri, 15 Aug 2025 00:36:08 +0000
Subject: [PATCH] Set mDataSize properly in appendFrom

mDataSize only increases if we end up with mDataPos greater than
mDataSize.

Test: binder_parcel_fuzzer
Bug: 438098181
Flag: EXEMPT bug fix
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:ab883e120ae18ef93a4c6f17c05a14218da5118e)
Merged-In: I932ad50627d15452f67b39bd410d666dc8e70a42
Change-Id: I932ad50627d15452f67b39bd410d666dc8e70a42
---
 libs/binder/Parcel.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libs/binder/Parcel.cpp b/libs/binder/Parcel.cpp
index 32d4d2959f..8b5a4cdc38 100644
--- a/libs/binder/Parcel.cpp
+++ b/libs/binder/Parcel.cpp
@@ -509,7 +509,7 @@ status_t Parcel::appendFrom(const Parcel* parcel, size_t offset, size_t len) {
     // append data
     memcpy(mData + mDataPos, data + offset, len);
     mDataPos += len;
-    mDataSize += len;
+    if (mDataPos > mDataSize) mDataSize = mDataPos;
 
     err = NO_ERROR;
 
-- 
2.52.0.rc1.455.g30608eb744-goog

